Source/Scene/I3SDataProvider.js

```diff
@@ -0,0 +1,1322 @@
+I3SSceneLayer.prototype._computeExtent = function () {
+  let minLongitude = Number.MAX_VALUE;
+  let maxLongitude = -Number.MAX_VALUE;
+  let minLatitude = Number.MAX_VALUE;
+  let maxLatitude = -Number.MAX_VALUE;
+
+  // Compute the extent from all layers
+  for (
+    let layerIndex = 0;
+    layerIndex < this._layerCollection.length;
+    layerIndex++
+  ) {
+    if (
+      (this._layerCollection[layerIndex]._data.store &&
+        this._layerCollection[layerIndex]._data.store.extent) ||
+      this._layerCollection[layerIndex]._data.fullExtent
+    ) {
+      const layerExtent = this._layerCollection[layerIndex]._data.fullExtent
+        ? this._layerCollection[layerIndex]._data.fullExtent
+        : this._layerCollection[layerIndex]._data.store.extent;
+      if (layerExtent && this._layerCollection[layerIndex]._data.fullExtent) {
+        minLongitude = Math.min(minLongitude, layerExtent.xmin);
+        minLatitude = Math.min(minLatitude, layerExtent.ymin);
+        maxLongitude = Math.max(maxLongitude, layerExtent.xmax);
+        maxLatitude = Math.max(maxLatitude, layerExtent.ymax);
+      } else if (
+        layerExtent &&
+        this._layerCollection[layerIndex]._data.store.extent
+      ) {
+        minLongitude = Math.min(minLongitude, layerExtent[0]);
+        minLatitude = Math.min(minLatitude, layerExtent[1]);
+        maxLongitude = Math.max(maxLongitude, layerExtent[2]);
+        maxLatitude = Math.max(maxLatitude, layerExtent[3]);
+      }
+    }
+    this._extent = computeExtent(
+      minLongitude,
+      minLatitude,
+      maxLongitude,
+      maxLatitude
+    );
+  }
+};
+/**
+ * This class implements an I3S layer, in Cesium, each I3SLayer
+ * creates a Cesium3DTileset
+ * @private
+ * @alias I3SLayer
+ * @constructor
+ * @param {I3SSceneLayer} [sceneLayer] The scene layer
+ * @param {object} [layerData] The layer data that is loaded from the scene
+ * layer
+ * @param {number} [index] The index of the layer to be reflected
+ */
+function I3SLayer(sceneLayer, layerData, index) {
+  this._parent = sceneLayer;
+  this._dataProvider = sceneLayer._dataProvider;
+
+  if (layerData.href === undefined) {
+    // assign a default layer
+    layerData.href = `./layers/${index}`;
+  }
+
+  this._uri = layerData.href;
+  let query = "";
+  if (this._dataProvider._query && this._dataProvider._query !== "") {
+    query = `?${this._dataProvider._query}`;
+  }
+  let tilesetUrl = "";
+  if (`${sceneLayer._dataProvider._url}`.match(/layers\/\d/)) {
+    tilesetUrl = `${sceneLayer._dataProvider._url}`.replace(/\/+$/, "");
+  } else {
+    // Add '/' to url if needed + `${this._uri}` if tileseturl not already in ../layers/[id] foramt
+    tilesetUrl = `${sceneLayer._dataProvider._url}`
+      .replace(/\/?$/, "/")
+      .concat(`${this._uri}`);
+  }
+  this._completeUriWithoutQuery = tilesetUrl;
+  this._completeUri = this._completeUriWithoutQuery + query;
+  this._data = layerData;
+  this._entities = {};
+  this._rootNode = null;
```

Comment:
Typically we use `undefined` instead of `null` to indicate that a variable is unset.

It looks like it's safe to change `null` -> `undefined` everywhere in the I3S files. Does that seem ok?